OpenRA.Mods.Common/OpenRA.Mods.Common.csproj

```diff
@@ -37,6 +37,9 @@
     <Optimize>true</Optimize>
   </PropertyGroup>
   <ItemGroup>
+    <Reference Include="BeaconLib">
+      <HintPath>..\thirdparty\download\BeaconLib.dll</HintPath>
```

Comment:
You'll need to update the scripts in /thirdparty for this.  You will also need to update the makefile so that this will compile on non-windows systems.